server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -856,11 +855,13 @@ public VolumeVO doInTransaction(TransactionStatus status) {
         });
     }
 
+    @Override
     public boolean validateVolumeSizeRange(long size) {
+        long maxVolumeSize = VolumeOrchestrationService.MaxVolumeSize.value();
         if (size < 0 || (size > 0 && size < (1024 * 1024 * 1024))) {
             throw new InvalidParameterValueException("Please specify a size of at least 1 GB.");
-        } else if (size > (_maxVolumeSizeInGb * 1024 * 1024 * 1024)) {
-            throw new InvalidParameterValueException("Requested volume size is " + size + ", but the maximum size allowed is " + _maxVolumeSizeInGb + " GB.");
+        } else if (size > (maxVolumeSize * 1024 * 1024 * 1024)) {
+            throw new InvalidParameterValueException(String.format("Requested volume size is %d, but the maximum size allowed is %d GB.", size, maxVolumeSize));
```

Comment:
@shwstppr Can you update the log so that both the sizes are in sync with the units. For ex:

_Requested volume size is 21474836480, but the maximum size allowed is 15 GB._  => `Requested volume size is 20 GB, but the maximum size allowed is 15 GB.`

Reply:
@sureshanaparti I've made the change to fix this.
I see you've started a new test run. I don't think this change affects results. Up to you.

Reply:
> @sureshanaparti I've made the change to fix this.
> I see you've started a new test run. I don't think this change affects results. Up to you.

if not further changes, would upgrade the package and verify this log.